pkg/dev/podmandev/cleanup.go

```diff
@@ -4,12 +4,24 @@ import (
 	"context"
 	"fmt"
 	"io"
+
+	corev1 "k8s.io/api/core/v1"
+
+	odocontext "github.com/redhat-developer/odo/pkg/odo/context"
+	"github.com/redhat-developer/odo/pkg/util"
 )
 
 func (o *DevClient) CleanupResources(ctx context.Context, out io.Writer) error {
 	fmt.Printf("Cleaning up resources\n")
 	if o.deployedPod == nil {
-		return nil
+		compName := odocontext.GetComponentName(ctx)
+		appName := odocontext.GetApplication(ctx)
+		name, err := util.NamespaceKubernetesObject(compName, appName)
+		if err != nil {
+			return nil
+		}
+		o.deployedPod = &corev1.Pod{}
+		o.deployedPod.SetName(name)
```

Comment:
Is this related to this PR? What is the problem fixed by this change?

Reply:
In case of some failures, pod is still created, this change attempts at fixing this problem.

Reply:
I would prefer to fix the problem why deployedPod is null when a Pod is actually deployed

Reply:
Podman creates the pod even if something is incorrect with the manifest, for e.g. non-existent image.